app/models/user.rb

```diff
@@ -0,0 +1,36 @@
+require 'bcrypt'
+
+class User
+
+  include DataMapper::Resource
+
+  has n, :messages
+
+  property :id, Serial
+  property :email, String, unique: true, message: 'This email is already taken'
+  property :username, String
+  property :name, String
+  property :password_digest, Text
+
+
+  attr_reader :password
+  attr_accessor :password_confirmation
+
+  validates_uniqueness_of :email
+  validates_confirmation_of :password
```

Comment:
Prefer the new style validations `validates :column, confirmation: value` over `validates_confirmation_of`.